conformance/scenarios/httpgateway/httpgateway_test.go

```diff
@@ -0,0 +1,114 @@
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package httpgateway
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"testing"
+	"time"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"sigs.k8s.io/service-apis/conformance/utils"
+)
+
+func TestHttpGateway(t *testing.T) {
+	// Test Setup
+	c, sc, err := utils.LoadClientset()
+	if err != nil {
+		t.Fatalf("error loading clientset: %v", err)
+	}
+	err = utils.DynamicApply(utils.DynamicParams{Path: "../../../config/crd/bases"})
+	if err != nil {
+		t.Fatalf("error installing service-apis CRDs: %v", err)
+	}
+	ns, err := utils.NewNamespace(c)
+	if err != nil {
+		t.Fatalf("error creating namespace: %v", err)
+	}
+
+	controllerName := os.Getenv("CONTROLLER_NAME")
+	if controllerName == "" {
+		t.Fatalf("CONTROLLER_NAME environment variable must be set")
+	}
+
+	dp := utils.DynamicParams{
+		Path:      "httpgateway.yaml",
+		Namespace: ns,
+		Params:    map[string]string{"Controller": controllerName},
+	}
+	err = utils.DynamicApply(dp)
+	if err != nil {
+		t.Fatalf("error installing service-apis examples: %v", err)
+	}
+
+	// Test cases
+	t.Run("GatewayClass.Status Admitted condition should be set to true within 1 minute", func(t *testing.T) {
+		err := wait.PollImmediate(5*time.Second, 1*time.Minute, func() (bool, error) {
+			gwc, err := sc.NetworkingV1alpha1().GatewayClasses().Get(context.TODO(), "acme-lb", metav1.GetOptions{})
+			if err != nil {
+				return false, fmt.Errorf("error fetching GatewayClass: %v", err)
+			}
+
+			for _, cond := range gwc.Status.Conditions {
+				if cond.Type == "Admitted" && cond.Status == "True" {
+					return true, nil
+				}
+			}
+
+			return false, nil
+		})
+		if err != nil {
+			t.Errorf("GatewayClass.Status Admitted condition was not set to true: %v", err)
+		}
+	})
+
+	t.Run("Gateway.Status Scheduled condition should be set to true within 3 minutes", func(t *testing.T) {
+		err := wait.PollImmediate(5*time.Second, 3*time.Minute, func() (bool, error) {
+			gwc, err := sc.NetworkingV1alpha1().Gateways(ns).Get(context.TODO(), "my-gateway", metav1.GetOptions{})
+			if err != nil {
+				return false, fmt.Errorf("error fetching Gateway: %v", err)
+			}
+
+			for _, cond := range gwc.Status.Conditions {
+				if cond.Type == "Scheduled" && cond.Status == "True" {
+					return true, nil
+				}
+			}
+
+			return false, nil
+		})
+		if err != nil {
+			t.Errorf("Gateway.Status Scheduled condition was not set to true: %v", err)
+		}
+	})
+
+	// Cleanup
+	t.Cleanup(func() {
```

Comment:
is t.Cleanup() needed here ? As far as I know, calling the function means it will run at the end of test execution - but this is already the end, so we could just run the function directly?

Please let me know if I am wrong, I commented the same on some other PR and don't want to spread nonsense everywhere :slightly_smiling_face:

Reply:
Yep, you're completely right. I added it to be a bit more explicit about what was happening, but it's probably overkill when combined with the comment above.